### PR TITLE
feat: migrate to yargs

### LINKED ISSUE #86: 🚀 Feature: Switch from optimist (deprecated) to yargs
We don't need to get into a conversation about whether [warnings like this](https://github.com/altano/npm-packages/security/dependabot/106) are helpful or not, but they are happening.

My latest one was from pulling in `eslint-plugin-package-json`, because of this dependency chain:

```
eslint-plugin-package-json 0.12.0
└─┬ package-json-validator 0.6.3
  └─┬ optimist 0.6.1
    └── minimist 0.0.10
```

It looks like package-json-validator would benefit from either moving from optimist (deprecated) to directly using an updated version of minimist or switching to something like yargs.

### PR DESCRIPTION
**Pull request by de-don**
closes #86

### CODE REVIEW

**Comment by de-don** [comment]
@JoshuaKGoldberg looks like CI is not configured properly :(

**Comment by Cellule** [comment]
@JoshuaKGoldberg Any chance we can get this merged in?

**Comment by Cellule** [comment]
> @JoshuaKGoldberg looks like CI is not configured properly :(

Because the pipeline are defined in https://github.com/JoshuaKGoldberg/package.json-validator/pull/89
Normally you shouldn't put a pipeline required in the settings until it has been merged in

**Comment by JoshuaKGoldberg** [comment]
Ha, yeah, I'd run [`create-typescript-app`](http://github.com/JoshuaKGoldberg/create-typescript-app) and meant to merge, then got pulled away for other things. I should be able to get to #89 and this Soon™️. Sorry for the delay!

File: bin/pjv [hunk 1]
```diff
@@ -7,46 +7,34 @@
...
-	.describe("help", "this help message")
-	.alias("help", "?");
-
-var options = args.argv;
-
-if (options.help) {
-	args.showHelp();
-	process.exit(0);
-}
+	.parse();
```
Comment (JoshuaKGoldberg) [inline bin/pjv:40]: 🤔 Am I doing something wrong? When I put in a `console.log({ options })` here, `recommendations` and `warnings` aren't detected:

```plaintext
joshgoldberg ~/repos/package.json-validator $ node bin/pjv -r -w   
{
  options: {
    _: [],
    r: false,
    recommendations: false,
    w: false,
    warnings: false,
    filename: 'package.json',
    f: 'package.json',
    spec: 'npm',
    s: 'npm',
    quiet: false,
    q: false,
    '$0': 'bin/pjv'
  }
}
{ valid: true }
joshgoldberg ~/repos/package.json-validator $ node bin/pjv --recommendations --warnings
{
  options: {
    _: [],
    recommendations: false,
    r: false,
    warnings: false,
    w: false,
    filename: 'package.json',
    f: 'package.json',
    spec: 'npm',
    s: 'npm',
    quiet: false,
    q: false,
    '$0': 'bin/pjv'
  }
}
```

They are on `main`.

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
Sorry again for the wait here. I don't think it's working? Am I missing something?

**Comment by de-don** [comment]
@JoshuaKGoldberg sorry, I forgot to mark these options as boolean. Should be fixed

**Review by JoshuaKGoldberg — APPROVED** [review]
💯 this is lovely, thanks @de-don!

**Comment by JoshuaKGoldberg** [comment]
@all-contributors please add @de-don for code.

> 🤖 Beep boop! This comment was added automatically by [all-contributors-auto-action](https://github.com/marketplace/actions/all-contributors-auto-action).
> Not all contributions can be detected from Git & GitHub alone. Please comment any missing contribution types this bot missed.
> ...and of course, thank you for contributing! 💙

**Comment by Cellule** [comment]
Awesome thanks!
I opened a PR to update to 0.7.0 in https://github.com/JoshuaKGoldberg/eslint-plugin-package-json/pull/569

**Pull request merged**